src/components/templates/ScreenShare/agoraHooks/useAgoraRemotes.ts

```diff
@@ -0,0 +1,47 @@
+import { IAgoraRTCClient, IAgoraRTCRemoteUser } from "agora-rtc-sdk-ng";
+import { useCallback, useEffect, useState } from "react";
+
+const useAgoraRemotes = (
+  client: IAgoraRTCClient | undefined,
+  channel: { appId: string; channel: string; token: string }
+) => {
+  const [remoteUsers, setRemoteUsers] = useState<IAgoraRTCRemoteUser[]>([]);
+
+  const updateRemoteUsers = useCallback(() => {
+    if (!client) return;
+    setRemoteUsers(() => Array.from(client.remoteUsers));
+  }, [client]);
+
+  const handleUserPublished = useCallback(
+    async (user: IAgoraRTCRemoteUser, mediaType: "audio" | "video") => {
+      if (!client) return;
+
+      await client.subscribe(user, mediaType);
+      updateRemoteUsers();
+    },
+    [client, updateRemoteUsers]
+  );
+
+  useEffect(() => {
+    if (!client) return;
```

Comment:
```suggestion
    if (!client) return;
    
```

Reply:
To explain this change, our pattern is to add a newline after these 'guard statements' to make the code easier to read.